src/module/weatherEffects/WeatherLayer.js

```diff
@@ -114,23 +135,34 @@ export class WeatherLayer extends CanvasLayer {
   async tearDown() {
     Object.values(this.weatherEffects).forEach(({ fx }) => fx.stop());
 
-    this.weather = null;
+    this.weather = undefined;
     this.mask = null;
```

Comment:
@dev7355608 Any idea what might cause the scene to crash when it is redrawn (e.g. due to the background image being changed) when this line is removed and PV is active? Maybe related to this change? https://github.com/dev7355608/perfect-vision/commit/5a1452d3184fdd38630cc21eebbc9ca83197decf

Reply:
Tear down destroys all objects, but doesn't clear the mask field of the layer. So once rendering continues it tries to use the destroyed mask, which fails horribly.

Reply:
But I never set a mask myself. I see that a mask is being set, probably by PV. But then it should also be PVs responsibility to set it back to `null` after it being destroyed.

Reply:
The cleanup of the mask will added to PV.